pkg/parse/namespace.go

```diff
@@ -271,14 +271,19 @@ func (p *namespace) setSyncStatusWithRetries(ctx context.Context, newStatus sync
 
 	setSyncStatusFields(&rs.Status.Status, newStatus, denominator)
 
-	errorSources, errorSummary := summarizeErrors(rs.Status.Source, rs.Status.Sync)
-	if newStatus.syncing {
-		reposync.SetSyncing(rs, true, "Sync", "Syncing", rs.Status.Sync.Commit, errorSources, errorSummary, rs.Status.Sync.LastUpdate)
-	} else {
-		if errorSummary.TotalCount == 0 {
-			rs.Status.LastSyncedCommit = rs.Status.Sync.Commit
+	// Only update the Syncing condition if we haven't already fetched a new commit.
```

Comment:
What are the scenarios we're trying to protect against here? Could you add a bit more context to the comment?

Reply:
If there's been a new commit, it's ok for the new commit fetch status to override the old fetch/parse status. But if it's an old commit, there could be parse errors we don't want to erase. This is necessary because the fetch and parse status both share the status.source fields. 

A better fix long term would be to give the fetch & parse phases different status fields, but that should have been done when rendering was added and wasn't. So now it's API debt. Fixing it would require also changing the CLI & UI. 

@nan-yu may end up needing to do this as a blocker for hooks. But for now the API is left as-is with slightly different behavior.

Reply:
We also don't need to update the last updated timestamp or syncing condition if the commit hasn't changed. So we can just bypass the whole update. 

It's hypothetically possible that parsing succeeds after previously failing, but I can't think of many cases where this would happen without a new commit. Maybe a disk read error? If that's the case, the error will eventually be removed when parsing succeeds or fails, which is how it works today already.

Reply:
Updated the comment with more context.

Reply:
To clarify - the reason for all of this complexity is because of the remediator status updates, right? It seems like in the absence of the remediator status updates, all of the status updates would be sequential and we wouldn't need to compare commits.

Reply:
Correct. Having the remediator run in parallel to the rest of the pipeline makes this complicated. And having remediator errors share the same status fields & conditions makes it worse.